File: regexlib/2021-5-15/javascript_test_file/regexlib_6153.js
/* 6153
 * ([^_.]([a-zA-Z0-9_]*[.]?[a-zA-Z0-9_]+[^_]){2})@([a-z0-9]+[.]([a-z]{2,3}|[a-z]{2,3}[.][a-z]{2,3}))
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:"@a@a@"+"0000"*32+"◎1 __EOA(iii)"
 */
var REGEX = /([^_.]([a-zA-Z0-9_]*[.]?[a-zA-Z0-9_]+[^_]){2})@([a-z0-9]+[.]([a-z]{2,3}|[a-z]{2,3}[.][a-z]{2,3}))/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '@a@a@' + '0000'.repeat(i*1) + '◎1 __EOA(iii)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}

File: regexlib/python_re_test_file/regexlib_6153.py
# 6153
# ([^_.]([a-zA-Z0-9_]*[.]?[a-zA-Z0-9_]+[^_]){2})@([a-z0-9]+[.]([a-z]{2,3}|[a-z]{2,3}[.][a-z]{2,3}))
# EXPONENT
# nums:4
# EXPONENT AttackString:"@a@a@"+"0000"*32+"◎1 __EOA(iii)"

import re
from time import perf_counter

regex = """([^_.]([a-zA-Z0-9_]*[.]?[a-zA-Z0-9_]+[^_]){2})@([a-z0-9]+[.]([a-z]{2,3}|[a-z]{2,3}[.][a-z]{2,3}))"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "@a@a@" + "0000" * i * 1 + "◎1 __EOA(iii)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")